src/modules/ie-objects/ObjectDetailPage.tsx

```diff
@@ -802,19 +802,22 @@ export const ObjectDetailPage: FC<DefaultSeoInfo> = ({ title, description, image
 		[metadataExportDropdownOpen, onExportClick, tHtml, tText]
 	);
 
-	const highlighted = (toHighlight: string) => (
-		<Highlighter
-			searchWords={
-				query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS]
-					? decodeURIComponent(
-							query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS] as string
-					  ).split(HIGHLIGHTED_SEARCH_TERMS_SEPARATOR)
-					: []
-			}
-			autoEscape={true}
-			textToHighlight={toHighlight}
-		/>
-	);
+	const highlighted = (toHighlight: string, searchTerms?: string[]) => {
+		const searchWords = searchTerms
+			? searchTerms
+			: query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS]
+			? decodeURIComponent(query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS] as string).split(
+					HIGHLIGHTED_SEARCH_TERMS_SEPARATOR
+			  )
+			: [];
```

Comment:
Note that `|| []` does not make `!searchTerms` redundant

```suggestion
		const param = query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS];
		let searchWords = searchTerms || [];

		if (!searchTerms && param) {
			searchWords = decodeURIComponent(param).split(HIGHLIGHTED_SEARCH_TERMS_SEPARATOR);
		}
```

Reply:
i'm not sure i follow what you're trying to say

the code should be as follows:
* use searchTerms if they exist
* use query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS] if they exist
* use [] if all else fails

Maybe it's clearer if i write it using if else instead of a double tertiary operator:
```typescript
		let searchWords: string[];
		if (searchTerms) {
			searchWords = searchTerms;
		} else if (query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS]) {
			searchWords = decodeURIComponent(query[QUERY_PARAM_KEY.HIGHLIGHTED_SEARCH_TERMS] as string).split(
				HIGHLIGHTED_SEARCH_TERMS_SEPARATOR
			);
		} else {
			searchWords = [];
		}
```